templates/typescript/ab_testing.ts

```diff
@@ -0,0 +1,25 @@
+async function handleRequest(request) {
+  const NAME = 'experiment-0'
+  // The Responses below are placeholders. You can set up a custom path for each test (e.g. /control/somepath ).
+  const TEST_RESPONSE = new Response('Test group') // e.g. await fetch('/test/sompath', request)
+  const CONTROL_RESPONSE = new Response('Control group') // e.g. await fetch('/control/sompath', request)
+  // Determine which group this requester is in.
+  const cookie = request.headers.get('cookie')
+  if (cookie && cookie.includes(`${NAME}=control`)) {
+    return CONTROL_RESPONSE
+  } else if (cookie && cookie.includes(`${NAME}=test`)) {
+    return TEST_RESPONSE
+  } else {
+    // If there is no cookie, this is a new client. Choose a group and set the cookie.
+    let group = Math.random() < 0.5 ? 'test' : 'control' // 50/50 split
+    let response = group === 'control' ? CONTROL_RESPONSE : TEST_RESPONSE
+    response.headers.append('Set-Cookie', `${NAME}=${group}; path=/`)
+    return response
+  }
+}
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+
+export {}
```

Comment:
what's going on with all these empty exports?

Reply:
Discussed separately in chat, but the tl;dr is:
* If there's no `export` statement, TypeScript assumes all the `handleRequest` functions are being defined in the global scope, and throws an error for duplicate function definitions.
* If I do something more sensible like `export async function handleRequest...` or `export { handleRequest }` in the TypeScript version, that gets transpiled directly into the JavaScript version which we present to users – which is bad, because that causes syntax errors if you drop the snippets into the playground.

@1000hz and I discussed other options like having a tsconfig/project per snippet, but decided that was overkill and would make the experience of contributing to this repo worse, and that this is OK for now.